packages/@react-native-windows/telemetry/src/telemetry.ts

```diff
@@ -4,207 +4,411 @@
+    | projectUtils.AppProjectInfo
+    | projectUtils.DependencyProjectInfo = undefined;
+
+  protected static reset(): void {
+    // Reset client
     if (Telemetry.client) {
-      Telemetry.client.config.disableAppInsights = true;
+      Telemetry.client.flush();
+      Telemetry.client = undefined;
     }
-    Telemetry.shouldDisable = true;
+
+    // Reset local members
+    Telemetry.options = {
+      setupString: DefaultSetupString,
+      preserveErrorMessages: false,
+    };
+    Telemetry.commandInfo = {};
+    Telemetry.versionsProp = {};
+    Telemetry.projectProp = undefined;
+  }
+
+  static isEnabled(): boolean {
+    return Telemetry.client !== undefined;
   }
 
-  static setup(preserveMessages?: boolean) {
-    if (Telemetry.isCI()) {
-      this.disable();
+  static getSessionId(): string {
+    return basePropUtils.getSessionId();
+  }
+
+  /** Sets up the Telemetry static to be used elsewhere. */
+  static async setup(options?: Partial<TelemetryOptions>) {
+    if (Telemetry.client) {
+      // Bail since we've already setup
       return;
     }
-    if (Telemetry.client) {
+
+    // Bail if we're in CI and not capturing CI
+    if (!this.isTest && basePropUtils.isCI() && !basePropUtils.captureCI()) {
       return;
     }
-    if (!process.env.RNW_CLI_TEST) {
-      appInsights.Configuration.setInternalLogging(false, false);
+
+    // Save off options for later
+    Object.assign(Telemetry.options, options);
+
+    Telemetry.setupClient();
+
+    await Telemetry.setupBaseProperties();
+
+    Telemetry.setupTelemetryProcessors();
+  }
+
+  /** Sets up Telemetry.client. */
+  private static setupClient() {
+    appInsights.Configuration.setInternalLogging(
+      Telemetry.isTest,
+      Telemetry.isTest,
+    );
+
+    Telemetry.client = new appInsights.TelemetryClient(
+      Telemetry.options.setupString,
+    );
+
+    // Uncomment for Fiddler testing
+    // Telemetry.client.config.proxyHttpUrl = 'http://localhost:8888';
+    // Telemetry.client.config.proxyHttpsUrl = 'http://localhost:8888';
+
+    Telemetry.client.config.disableAppInsights = Telemetry.isTest;
+    Telemetry.client.channel.setUseDiskRetryCaching(!Telemetry.isTest);
+  }
+
+  /** Sets up any base properties that all telemetry events require. */
+  private static async setupBaseProperties() {
+    Telemetry.client!.commonProperties.deviceId = await basePropUtils.deviceId();
+    Telemetry.client!.commonProperties.deviceLocale = await basePropUtils.deviceLocale();
+    Telemetry.client!.commonProperties.deviceNumCPUs = basePropUtils
+      .deviceNumCPUs()
+      .toString();
+    Telemetry.client!.commonProperties.deviceTotalMemory = basePropUtils
+      .deviceTotalMemory()
+      .toString();
+    Telemetry.client!.commonProperties.deviceDiskFreeSpace = basePropUtils
+      .deviceDiskFreeSpace()
+      .toString();
+    Telemetry.client!.commonProperties.ciCaptured = basePropUtils
+      .captureCI()
+      .toString();
```

Comment:
Will this ever be `false` when `ciType` is also set?

Reply:
We are currently planning on _not_ capturing telemetry from CI runs. However if we decided to change that decision, it would be akin to changing our event sampling rate.

Having this field captured here lets us partition the data on its value and calculate accurate statistics.